include/glow/Optimizer/GraphOptimizer/PassManager.h

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef GLOW_OPTIMIZER_GRAPHOPTIMIZER_PASSMANAGER_H
+#define GLOW_OPTIMIZER_GRAPHOPTIMIZER_PASSMANAGER_H
+
+#include "glow/Optimizer/GraphOptimizer/CompilationContext.h"
+#include "glow/Optimizer/GraphOptimizer/FunctionPass.h"
+#include "glow/Optimizer/GraphOptimizer/FunctionPasses.h"
+#include "glow/Optimizer/GraphOptimizer/Pipeline/Pipeline.h"
+
+#include "llvm/ADT/SmallVector.h"
+
+namespace glow {
+
+/// Manager for running a series of FunctionPasses. Given some Function,
+/// CompilationContext, and provided Pipeline, it will run all passes on the
+/// Function. Enables easier debugging given runPrePass() and runPostPass()
+/// calls, which can be modified to run some code before or before every pass.
+class FunctionPassManager : public Named {
```

Comment:
Should most of the methods in `FunctionPassManager` be virtual?

Reply:
In terms of allowing future pass managers to inherit from this?

I was thinking in a future PR I would create an abstract PassManager class with these methods as virtual, and then the FunctionPassManager/ModulePassManager/IRFunctionPassManager/etc. would implement them. But that requires some work on potentially adding some IRUnit class for Module/Function/Node/Instruction to inherit from, or templatizing it, and so thought we would do that in future PRs.

Reply:
> In terms of allowing future pass managers to inherit from this?

Yes. It could happen even for Function-based PassManagers, e.g. if you want to redefine pre/post processing somewhat or something like that. But this is a very minor comment. Feel free to ignore for now.